apps/timetable/admin/ui/js/index.js

```diff
@@ -0,0 +1,411 @@
+/*!
+ * Copyright 2014 Digital Services, University of Cambridge Licensed
+ * under the Educational Community License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['gh.core', 'bootstrap.calendar', 'bootstrap.listview', 'chosen', 'clickover', 'jquery-bbq'], function(gh) {
+
+    var triposData = {
+        'courses': [],
+        'subjects': [],
+        'parts': [],
+        'modules': []
+    };
+
+
+    /////////////////
+    //  RENDERING  //
+    /////////////////
+
+    /**
+     * Get the tripos structure from the REST API and filter it down for easy access in the templates
```

Comment:
We should probably make a utility out of this as we have similar or the same functionality for the student UI. could be something like `getTriposStructure` that deals with all of this and returns the `triposData` object.